docs/developers/Firmware/encryption.mdx

```diff
@@ -52,3 +52,63 @@ I'm assuming that meshtastic is being used to hike in places where someone capab
 - Two other things to keep in mind are that AES-CTR does not itself provide authenticity (e.g. an attacker can flip bits in replaying data and scramble the resulting plaintext), and that the current scheme gives some hints about transmission in the size. So, if you worry about an adversary deliberately messing-up messages or knowing the length of a text message, it looks like those might be possible.
 
 I'm guessing that the network behaves somewhat like a store-and-forward network - or, at least, that the goal is to avoid establishing a two-way connection to transmit data. I'm afraid I haven't worked with mesh networks much, but remember studying them briefly in school about ten years ago.
+
+
+## Phased Proposal for the Meshtastic Security Framework
+
+### Phase 1 - Fixed network encryption with AES-CTR
+
+The current implementation provides optional confidentiality to members of a configured network:
+ * Encryption is implemented in devices/nodes with network-wide encryption keys.
+ * Encryption is optional and is turned off when devices are in 'Ham mode'.
+ * There is no encryption supported in the clients (iOS, Android) to facilitate distribution as mass market software.
+ * Pairing from client-to-device is by:
+    * direct USB cable
+    * BT pairing
+ * Devices are 'promiscuous' and will pair with any near-by client. Network confidentiality requires physical protextion of all nodes.
+
+### Phase 2 - Strong device and client identity
+
+**Phase 2 security goals:**
+ * Know who sent a message (strong binding of messages to a particular node and/or terminal device)
+    * This would be an optional feature for a message
+ * Optionaly enforce identity based restrictions on some actions performed at nodes and/or clients
+ * Optional support of strong pairing of a client to a device/node and restrict ability to manage and receive messages based on the pairing.
+    * The BT paring and the cryptographic paring are separate (to simplify pahse 1 deployment and testing)
+ * Above features should be architected to be ‘cryptographically strong’ and algorithm agile.
+
+**Phase 2 Proposed mechanisms:**
+ * Proposed initial algorithms
+    - Ed25519 for signatures based on NaCl libraries and iOS support for Ed25519
+ * Clients and nodes to generate local identity Ed25519 keys 
+ * Devices maintain knowledge of owner public key
+ * Devices maintain knowledge of some peers public keys and associated information (name, etc.)
+ * Experimental protobuf message type with
+     * cipher suite indicator (csi)
+     * wrapped data using a cipher suite identifier to indicate use of Ed25519 wrapping identified algorithms. 
+       Wrapped data to contain any of the existing message types.
+     * initial ‘cipher suite’ **only** signs a message
```

Comment:
I think the concept of a ‘cipher suite’ should be limited to alg selection and not cover anything to do with signing

Reply:
... the first one or more bytes of a protected message (encrypted, signed, whatever) should provide everything necessary to apply the cryptographic processing. Long term this should be designed to enable MAC based integrity, key ratcheting etc.  
On a slightly related topic ... on how to determine the cryptographic process...
The cryptographic processing should use other header fields. The node addresses can be used to determine the keys/session to use. Encrypted messages should be using an AEAD to protect the header fields.
I believe the 'only sign' for now might mitigate concerns about describing this work for export review.
This discussion also overlaps with the 'generic encrypt' versus 'protect' versus 'signed' message types.
Similar concepts were debated in IPSec for the ESP and AH modes of operation.  AH was created to provide integrity only (MAC based, not signed). It was barely used and could have been handled by ESP cipher negotiation. In the case of the Meshtastic environment. It would be useful to clearly be able to identify encrypted messages versus signed messages for 'Ham mode'